### ISSUE TITLE
Can't navigate when layout has more than one island

### ISSUE THREAD

**Issue by adamzerner** [denoland/fresh#1992]
See https://github.com/adamzerner/nav-issue-demo for reproduction.

First I created a new Fresh project. Then I set up a `_layout.tsx` file like so:

```
import { LayoutProps } from "$fresh/server.ts";
import { Foo } from "../islands/foo.tsx";
import { Bar } from "../islands/bar.tsx";

export default ({ Component }: LayoutProps) => {
  return (
    <div>
      <Foo />
      <Bar />
      <Component />
    </div>
  );
};
```
Where `Foo` and `Bar` are the following (no JavaScript):
```
export const Foo = () => {
  return <div>Foo</div>;
};
```
and
```
export const Bar = () => {
  return <div>Bar</div>;
};
```
Then, in `routes/_index.tsx`, I added `<a href="/greet/alice">Greet Alice</a>`. If I remove `<Bar />`, the link works. But with both `<Foo />` and `<Bar />` in the layout, it does not.

**Comment by deer** [comment]
Hey @adamzerner, thanks for including a reproduction repo -- that's really great. Unfortunately I can't seem to reproduce the issue! I click the link and see:
```
Foo
Bar
Hello alice
```
What does `deno --version` say for you? I get:
```
deno 1.37.2 (release, x86_64-apple-darwin)
v8 11.8.172.13
typescript 5.2.2
```

**Comment by adamzerner** [comment]
@deer Thanks!

Sorry about the issues reproducing. I just tried `rm -rf`ing the repo locally and then `git clone`ing to see if I could reproduce it after a `git clone`. I have had weird, inconsistent results that I'm still trying to pinpoint. For the time being, if you try some combination restarting your dev server a few times, refreshing the page, and navigating back in the browser after hitting `http://localhost:8000/greet/alice`, do you ever run into it?

```
code/nav-issue-demo [master] » deno --version
deno 1.37.2 (release, aarch64-apple-darwin)
v8 11.8.172.13
typescript 5.2.2
```

**Comment by adamzerner** [comment]
Hm, actually this seems to be a way to reproduce it. In short, after cloning, start the dev server, don't visit the site, kill the dev server, start the dev server again, visit the site, try clicking the link, and it won't work.
```
~/code » rm -rf nav-issue-demo
~/code » git clone git@github.com:adamzerner/nav-issue-demo.git
Cloning into 'nav-issue-demo'...
remote: Enumerating objects: 37, done.
remote: Counting objects: 100% (37/37), done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 37 (delta 4), reused 37 (delta 4), pack-reused 0
Receiving objects: 100% (37/37), 11.65 KiB | 2.33 MiB/s, done.
Resolving deltas: 100% (4/4), done.
~/code » cd nav-issue-demo
code/nav-issue-demo [master] » deno task start
Task start deno run -A --watch=static/,routes/ dev.ts
Watcher Process started.
The manifest has been generated for 6 routes and 3 islands.

 🍋 Fresh ready
    Local: http://localhost:8000/

^C
code/nav-issue-demo [master] » deno task start
Task start deno run -A --watch=static/,routes/ dev.ts
Watcher Process started.
The manifest has been generated for 6 routes and 3 islands.

 🍋 Fresh ready
    Local: http://localhost:8000/
Watcher File change detected! Restarting!

 🍋 Fresh ready
    Local: http://localhost:8000/
```

**Comment by deer** [comment]
Sadly I still can't reproduce it. I did try refreshing and restarting in various combinations as well as your suggested order.

**Comment by marvinhagemeister** [comment]
@adamzerner Are there any errors printed in the browser console or the network tab?

**Comment by adamzerner** [comment]
Hm, I think it might be specific to Firefox. I'm able to reproduce it in Firefox but not Chrome or Safari. I'm on the latest versions of Firefox, Chrome and Safari. 119.0, 119.0.6045.105, and 17.1 respectively. I'm also on macOS v14.1 (latest).

> Are there any errors printed in the browser console or the network tab?

I see this in my browser console:

```
TypeError: a.default.detectStore(...) is undefined
[h1-check.js:1:1301](moz-extension://278d97d8-a3f5-4940-a634-8dff07648ead/h1-check.js)
    u moz-extension://278d97d8-a3f5-4940-a634-8dff07648ead/h1-check.js:1
    1337 moz-extension://278d97d8-a3f5-4940-a634-8dff07648ead/h1-check.js:1
    n moz-extension://278d97d8-a3f5-4940-a634-8dff07648ead/h1-check.js:1
    <anonymous> moz-extension://278d97d8-a3f5-4940-a634-8dff07648ead/h1-check.js:1
    <anonymous> moz-extension://278d97d8-a3f5-4940-a634-8dff07648ead/h1-check.js:1
    inject resource://gre/modules/ExtensionContent.sys.mjs:581
    AsyncFunctionNext self-hosted:852
 Fresh  Connected to development server.
```

And nothing in my server console:

```
Watcher File change detected! Restarting!

 🍋 Fresh ready
    Local: http://localhost:8000/
```

There also aren't any issues in the network tab:
<img width="1440" alt="Screenshot 2023-11-02 at 12 23 41 PM" src="https://github.com/denoland/fresh/assets/3144254/ab810cde-1776-40a1-84ab-b0e2e3c93d97">

I was also able to reproduce it with my Firefox extensions disabled:
<img width="720" alt="Screenshot 2023-11-02 at 12 26 30 PM" src="https://github.com/denoland/fresh/assets/3144254/7c5ccee4-8249-4e42-bbd7-e4f3bd010b1b">

And in a private window (Firefox's version of Chrome's Incognito).

**Comment by darkship** [comment]
I have the same issue, it only happens on firefox with `deno task start`.

I can reproduce it with 
- create a project `deno run -A -r https://fresh.deno.dev` 
- run `deno task start`
- go to `http://localhost:8000/`
- change the url to `http://localhost:8000/greet/john` -> it sends you back to `http://localhost:8000/`
- remove  `<Counter count={count} />`
- change the url to `http://localhost:8000/greet/john` -> it works

Note: after removing `Counter`, the page doesn't refresh when changing it. That happens also on chrome.

In my private project, I don't use islands yet but the navigation is broken too. I see that in the terminal the right page is called but then it goes back

**Comment by marvinhagemeister** [comment]
I tried the steps in Firefox, but I'm unable to reproduce the described issue. Tried on macOS + windows with latest Firefox.

> Note: after removing Counter, the page doesn't refresh when changing it. That happens also on chrome.

This part is already fixed in `main`, see https://github.com/denoland/fresh/pull/1956

**Comment by darkship** [comment]
> I tried the steps in Firefox, but I'm unable to reproduce the described issue. Tried on macOS + windows with latest Firefox.

I should have said that I use Ubuntu (linux Mint)

> > Note: after removing Counter, the page doesn't refresh when changing it. That happens also on chrome.
> 
> This part is already fixed in `main`, see #1956

Nice! But it seams that it was not released yet. Last release was 3 weeks ago (1.5.2). The fix is 2 weeks ago.

Also I checked the latest Fresh version out, everything looks like it works there.

**Comment by marvinhagemeister** [comment]
Not sure what changed, but I'm able to reproduce it now. Investigating...
